syntax = "proto3";

package grpcTalk;

service ProfileService {
    // Create a profile
    rpc CreateUserProfile (CreateUserProfileRequest) returns (CreateUserProfileResponse);
    // Get the mentee or mentor details 
    rpc GetUserProfile (GetUserProfileRequest) returns (GetUserProfileResponse);
}


message User {
    // REQUIRED
    string firstName = 1;
    // REQUIRED
    string lastName = 2;
    // REQUIRED
    Term term = 3;
}


message Term {
    string season = 1;
    int32 year = 2;
} 


message Profile {
    // REQUIRED
    string profile = 1;
}


message CreateUserProfileRequest {
    User user = 1;
    Profile profile = 2;
}


message CreateUserProfileResponse {
    string status = 1;
}


message GetUserProfileRequest {
    User user = 1;
}


message GetUserProfileResponse {
    User user = 1;
    Profile profile = 2;
}
